trunk/src/graphics/paraview/geom2vtk.F90

```diff
@@ -135,6 +149,32 @@ subroutine geom2vtk(Sname,Sfile, IcE,IcN,IcP)
 !..Step 3 : Elements
 !
    ice_subd=0; icn_subd=0
+!..Additional Computational for computing the total number of elements
+!  (including subelements: vlevel > 0)
+!..VTU Format needs this information a-priori as VTU needs headers
+!  with this information before appending the data.
+   if(VIS_VTU) then
+      if(SECOND_ORDER_VIS) then
+         allocate(ELEM_TYPES(NRELES))
+         ELEM_TYPES = ZERO
+      else
+         do iel=1,NRELES
+            mdle = ELEM_ORDER(iel)
+
+            if (PARAVIEW_DOMAIN.ne.0) then
+               call find_domain(mdle, ndom)
+               if (ndom.ne.PARAVIEW_DOMAIN) cycle
+            endif
+
+            ntype = NODES(mdle)%ntype
+            call get_vis_nrelem(ntype, ivis)
+            ice_subd = ice_subd + ivis
+         enddo
+         allocate(ELEM_TYPES(ice_subd))
+         ELEM_TYPES = ZERO
```

Comment:
Why does `SECOND_ORDER_VIS` count elements differently? Does it not take `PARAVIEW_DOMAIN` into consideration?

Reply:
Because for second-order elements, there is no vlevel, hence the total number of elements in mesh and visualization stays the same, whereas for first-order, depending upon the vlevel we have childs corresponding to each mesh element.

Reply:
Oh, I see. I'd thought for some reason that `SECOND_ORDER_VIS` also supports `VLEVEL`...

Reply:
If the user has set `VLEVEL>0` and `SECOND_ORDER_VIS = .true.`, maybe we should print a warning to the user that upscaling via `VLEVEL` is only supported for linear element output (I can add that...)

Reply:
Once this commit goes through, I have routines for arbitrary order hexes and tests, which Dr. Demkowicz and I wrote. Needs some polishing and some query to paraview guys regarding prismatic elements; I can look into it rather than doing a vlevel on 2nd order elements.

Reply:
regarding PARAVIEW_DOMAIN: changes implemented

Reply:
Concerning vlevel with second order element: yes, we need to throw an error, and also, mixed meshes with pyramids should lead us to throw an error while visualization. But can we solve on a mixed mesh containing pyramids?

Reply:
The code should support hybrid meshes with pyramids, yes. They are needed to join hexa with tets, for example. However, I haven't actually computed any real problems using pyramids within a hybrid mesh, we'd need to probably do some testing.

Reply:
Issues concerning pyramids are moved to #135